File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int i, j;
        std::map<int,int> index;
        for (int c=0; c<nums.size(); c++) {
            int v = nums[c];
            int d = target - v;
            if (index.find(d) != index.end()) {
                if (index[d] <= c) {
                    i = index[d];
                    j = c;
                } else {
                    i = c;
                    j = index[d];
                } 
            } else {
                index[v] = c;
            }
        }
        return vector<int>{i, j};
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        R = len(numbers) - 1
        L = 0
        while True:
            total = numbers[R] + numbers[L]
            if total == target:
                break

            if total > target:
                R -= 1
            elif total < target:
                L += 1    
        
        return [L + 1, R + 1]